src/TitleTests.php

```diff
@@ -294,20 +294,22 @@ function check_variant_url_slugs($request_slug, $query) {
         global $wpdb;
 
         // Look up variant URL slugs that match the unknown request slug
-        $post_ids = $wpdb->get_col($wpdb->prepare("
-			SELECT post_id
-			FROM $wpdb->postmeta
-			WHERE meta_key LIKE 'title_variants_%_url_slug'
-			AND meta_value = '%s'
-		", $request_slug), 0);
-		if (empty($post_ids)) {
+        $post_id = $wpdb->get_var($wpdb->prepare("
+			SELECT pm.post_id
+			FROM $wpdb->postmeta AS pm, $wpdb->posts AS p
+			WHERE pm.meta_key LIKE 'title_variants_%_url_slug'
+			  AND pm.meta_value = '%s'
+              AND pm.post_id = p.ID
+              AND p.post_status = 'publish'
+		", $request_slug));
+		if (empty($post_id)) {
 			return [];
 		}
 
         // Replace the 'name' query with a list of IDs that have a variant slug
         $query_vars = $query->query_vars;
         unset($query_vars['name']);
-        $query_vars['post__in'] = $post_ids;
+        $query_vars['p'] = $post_id;
```

Comment:
Switch from a search for multiple post IDs to a single ID.

Reply:
This was motivated in part by reading the [source code](https://github.com/WordPress/wordpress-develop/blob/6.7/src/wp-includes/query.php#L729-L738) for `is_single()`, which calls `$wp_query->is_single()`, which checks for a few query params including `p`: https://github.com/WordPress/wordpress-develop/blob/6.7/src/wp-includes/class-wp-query.php#L874